frame/contracts/src/rent.rs

```diff
@@ -532,4 +323,301 @@ where
 
 		Ok((caller_code_len, tombstone_code_len))
 	}
+
+	/// Create a new `RentStatus` struct for pass through to a requesting contract.
+	pub fn rent_status(
+		free_balance: &BalanceOf<T>,
+		contract: &AliveContractInfo<T>,
+		aggregated_code_size: u32,
+		current_refcount: u32,
+		at_refcount: u32,
+	) -> RentStatus<T> {
+		let calc_share = |refcount: u32| {
+			aggregated_code_size.checked_div(refcount).unwrap_or(0)
+		};
+		let max_share = calc_share(1);
```

Comment:
Why not just `let max_share = aggregated_code_size;`?

Reply:
Yes. `max_share` is superflous.